### ISSUE TITLE
Validation loss problem

### ISSUE THREAD

**Issue by Fafa87** [wolny/pytorch-3dunet#45]
It seems to me that validation loss is not calculated correctly because:

In traning it is calculated without sigmoid (training mode).
In validation it is calculated with sigmoid (eval mode).

And you get this:
![image](https://user-images.githubusercontent.com/9865688/67456510-4ac89300-f631-11e9-8a06-084fd5a82651.png)

 
In validation eval mode is turned on where sigmoid is done always?

**Comment by wolny** [comment]
Hi @Fafa87,

great spot! It was indeed an error and as you've said the sigmoid (or softmax depending on the normalization used) was being applied twice when computing the loss in the validation phase: 1st by the forward pass (model was set to 'eval' mode so this line: https://github.com/wolny/pytorch-3dunet/blob/master/unet3d/model.py#L103 was being be executed) and 2nd by the loss function itself (since e.g. `CrossEntropyLoss` from PyTorch applies the softmax explicitly).
I fixed the issue in the last commit (see: https://github.com/wolny/pytorch-3dunet/commit/216431f2732e05e76194c418c9e851fd5221c817) by just mirroring the logic I've used in the `train` function where it was correctly computed.

Thanks again for reporting the issue!

**Comment by Fafa87** [comment]
@wolny I fixed it differently (first time with pytorch so I was hesitant to remove `self.model.eval()`) by adding override to models - now i see it was overkill :) 

I have chosen your framework for a base of my project so I will probably come back with more :)
At the moment I realised that you have significantly updated code - I will have to port these changes on mine (and make sure that in future it will be less painfull). One more thing that we are investigating is the speed of both predictions and training (e.g. we do not utilise NC6 card and it works as fast as on my laptops gpu) - if you have any hint I would appreciate it.
